skietbaan_fe/src/components/Login.js

```diff
@@ -46,70 +69,84 @@ class Login extends Component {
     let isValid = false;
     let stateUpdate = {
       invalidPassword: this.state.invalidPassword,
-      invalidUsername: this.state.invalidUsername
-     }
-    if (target.name === "passwordValue" ) {
-      stateUpdate.invalidPassword= false;
+      invalidUsername: this.state.invalidUsername,
+      usernameFound: true,
+      passwordFound: true
     }
-    if (target.name === "usernameValue" ) {
-      stateUpdate.invalidUsername= false;
-      isValid = false;
+    if (target.name === "passwordValue" && target.value.length > 0) {
+      stateUpdate.invalidPassword = false;
+    };
+    if (target.name === "usernameValue") {
+      stateUpdate.invalidUsername = true;
+      for (var i = 0; i < this.state.users.length; i++) {
+        if (this.state.users[i].username == target.value) {
+          stateUpdate.usernameFound = true;
+          stateUpdate.invalidUsername = false;
+          break;
+        }
+        stateUpdate.usernameFound = false;
+      }
+
     }
-    if(this.state.usernameValue && this.state.passwordValue){
+    if (this.state.usernameValue
+      && this.state.passwordValue
+      && !stateUpdate.invalidUsername) {
       isValid = true;
     }
     this.setState({
-      ...stateUpdate ,
+      ...stateUpdate,
       validForm: isValid
     });
   };
 
-  validate()
-  {
+  validate() {
     let isValid = true;
     let stateUpdate = {
       invalidPassword: false,
       invalidUsername: false
-     }
+    }
     if (this.state.passwordValue.length === 0) {
-      stateUpdate.invalidPassword= true;
+      stateUpdate.invalidPassword = true;
       isValid = false;
     };
     if (validateUsername(this.state.usernameValue)) {
-      stateUpdate.invalidUsername= true;
+      stateUpdate.invalidUsername = true;
       isValid = false;
     };
     this.setState({
-      ...stateUpdate ,
+      ...stateUpdate,
       validForm: isValid
     });
   }
 
   login() {
     this.validate();
     if (this.state.validForm) {
-      let sha1 = require('sha1');  
+      let sha1 = require('sha1');
       let hash = sha1(this.state.passwordValue);
       let RequestObject = {
         "Username": this.state.usernameValue,
         "Password": hash,
       }
-      fetch( URL +"/api/features/login", {
+      fetch(URL + "/api/features/login", {
         method: 'post',
         headers: {
           'Accept': 'application/json',
           'Content-Type': 'application/json'
         },
         body: JSON.stringify(RequestObject)
-      }).then(function(response) {
-        return response.json();
-      }).then( function(data) {
-        if(typeof data === "object")
-        {
-          document.cookie = "token =" +data.token+"; expires =Wed, 18 Dec 2030 12:00:00 UTC";
+      }).then(response => response.json()).then(data => {
+        if (typeof data === "object") {
+          document.cookie = "token =" + data.token + "; expires =Wed, 18 Dec 2030 12:00:00 UTC";
```

Comment:
Why is this cookie's lifespan so large 😂

Reply:
Once of login of sort !